simapp/simd/cmd/root.go

```diff
@@ -87,9 +88,10 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig) {
 
 	rootCmd.AddCommand(
 		genutilcli.InitCmd(simapp.ModuleBasics, simapp.DefaultNodeHome),
-		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
+		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome, staking.ValidateMsgInGenesis),
```

Comment:
This updates means changes to the initialization of the applciation, could be annoy to update, thoughts?

Reply:
@PaddyMc can you elaborate on what kinds of changes would be needed to the application initialization? This is a bit vague...

Reply:
Thanks for reviewing and sorry for the ambiguity, the application will start with the same commands but,

now when using the `CollectGenTxsCmd` and `GenTxCmd` cobra commands we need to pass in functions defined in the staking module

gentx => https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/simapp/simd/cmd/root.go#L93-L94 => `staking.AppModuleBasic{}`
collect-gentx => https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/simapp/simd/cmd/root.go#L91 => `staking.ValidateMsgInGenesis`

This means if this gets merged, downstream consumers of the sdk will have to update two lines of code in their `cmd/*d/main.go` file

This could be annoying for downstream consumers of the sdk, any thought on the devs? 🙇 

`e.g`
Have a function wrapper that defaults to the staking module functions but have a custom function exposed that allows custom inputs?

Then we would have two commands in the [`x/genutil/client/cli/gentx.go`](https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/x/genutil/client/cli/gentx.go#L45) file

1. GenTxCmd
1. GenTxCmdCustom

Where GenTxCmd would wrap GenTxCmdCustom and pass in the staking module functions 🤔 

This would allow downstream consumers to function as expected, while also giving the sdk flexibility to inject custom functions for different consensus modules

But maybe this isn't something that needs considering atm 

I've also updated the PR description to be more concise 🤝

Reply:
I've created a PR #7316 into this PR to illustrate what the :+1: